src/features/CourtCaseFilters/reducers/filters.ts

```diff
@@ -0,0 +1,156 @@
+import type { Filter, FilterAction } from "types/CourtCaseFilter"
+import { caseStateLabels } from "utils/caseStateFilters"
+
+const handleAddingFilters = (newState: Filter, action: FilterAction) => {
+  switch (action.type) {
+    case "urgency": {
+      newState.urgentFilter.value = action.value
+      newState.urgentFilter.label = action.value ? "Urgent" : "Non-urgent"
+      newState.urgentFilter.state = "Selected"
+      break
+    }
+    case "caseAge": {
+      if (newState.caseAgeFilter.filter((caseAgeFilter) => caseAgeFilter.value === action.value).length < 1) {
+        newState.caseAgeFilter.push({ value: action.value as string, state: "Selected" })
+      }
+      break
+    }
+    case "dateFrom": {
+      newState.dateFrom.value = action.value
+      newState.dateFrom.state = "Selected"
+      break
+    }
+    case "dateTo": {
+      newState.dateTo.value = action.value
+      newState.dateTo.state = "Selected"
+      break
+    }
+    case "caseState": {
+      newState.caseStateFilter.value = action.value
+      newState.caseStateFilter.label = caseStateLabels[action.value ?? ""]
+      newState.caseStateFilter.state = "Selected"
+      break
+    }
+    case "locked": {
+      newState.lockedFilter.value = action.value
+      newState.lockedFilter.label = action.value ? "Locked" : "Unlocked"
+      newState.lockedFilter.state = "Selected"
+      break
+    }
+    case "myCases": {
+      newState.myCasesFilter.value = action.value
+      newState.myCasesFilter.label = action.value ? "Cases locked to me" : undefined
+      newState.myCasesFilter.state = "Selected"
+      break
+    }
+    case "reason": {
+      // React might invoke our reducer more than once for a single event,
+      // so avoid duplicating reason filters
+      if (newState.reasonFilter.filter((reasonFilter) => reasonFilter.value === action.value).length < 1) {
```

Comment:
interesting that this is what was causing the problem, is it reproducible enough that we could add a cypress test for it?

Reply:
It was in the original reducer: https://github.com/ministryofjustice/bichard7-next-ui/pull/464/files#diff-88864ffabf9762a6197ac31ffe5d58b65eddafdda9162135bdbf303a9aea7208L64-L65

🤷‍♂️